### PR TITLE
Add Astro

### PR DESCRIPTION
**Pull request by FredKSchott**
Hey yall! This PR adds support for Astro (https://docs.astro.build) so that we can add Algolia to the docs site. 

Background: I'm the creator of Snowpack (https://snowpack.dev), and when I got set up with Algolia on that docs site my contact person at Algolia had said that I could add a future project here directly next time. Hope this is okay!

### CODE REVIEW

File: configs/astro.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "lvl1": ".content h1",
+    "lvl2": ".content h2",
+    "lvl3": ".content h3",
+    "lvl4": ".content h4",
+    "lvl5": ".content h5",
+    "text": ".content p, .content li"
+  },
+  "conversation_id": [
+    "1341434945"
```
Comment (FredKSchott) [inline configs/astro.json:23]: copied this from snowpack.json. Snowpack is my other project, so assuming this was okay to use for both.

`nb_hits` I wasn't sure about, and copied over as well from snowpack.dev

**Comment by shortcuts** [comment]
Hi @FredKSchott, 

Thanks for submitting your config.

Have you also applied? if not, please do so: https://docsearch.algolia.com/apply/ so I can merge the PR

File: configs/astro.json (hunk 1 above)
Comment (shortcuts) [reply configs/astro.json:23]: No need to re-apply, I'll keep this `conversation_id`. For the `nb_hits` it's mostly indicative of the first crawl but we don't keep track of it.

**Comment by shortcuts** [comment]
Hey @FredKSchott 

Here's the credentials:
- apiKey: 'c06007c9cb270acc40e1c7799bad8326'
- indexName: 'astro'

**Comment by FredKSchott** [comment]
@shortcuts thanks! I'm getting an error when I use this new API key:

```
{
    "name": "ApiError",
    "message": "Invalid Application-ID or API key",
    "status": 403,
    "transporterStackTrace": [
        {
            "request": {
                "data": "{\"requests\":[{\"indexName\":\"astro\",\"query\":\"ab\",\"params\":\"hitsPerPage=20&highlightPreTag=%3Cmark%3E&highlightPostTag=%3C%2Fmark%3E&attributesToRetrieve=%5B%22hierarchy.lvl0%22%2C%22hierarchy.lvl1%22%2C%22hierarchy.lvl2%22%2C%22hierarchy.lvl3%22%2C%22hierarchy.lvl4%22%2C%22hierarchy.lvl5%22%2C%22hierarchy.lvl6%22%2C%22content%22%2C%22type%22%2C%22url%22%5D&attributesToSnippet=%5B%22hierarchy.lvl1%3A10%22%2C%22hierarchy.lvl2%3A10%22%2C%22hierarchy.lvl3%3A10%22%2C%22hierarchy.lvl4%3A10%22%2C%22hierarchy.lvl5%3A10%22%2C%22hierarchy.lvl6%3A10%22%2C%22content%3A10%22%5D&snippetEllipsisText=%E2%80%A6\"}]}",
                "headers": {
                    "content-type": "application/x-www-form-urlencoded"
                },
                "method": "POST",
                "url": "https://BH4D9OD16A-dsn.algolia.net/1/indexes/*/queries?x-algolia-agent=Algolia%20for%20JavaScript%20(4.10.3)%3B%20Browser%20(lite)%3B%20docsearch%20(1.0.0-alpha.28)%3B%20docsearch-react%20(1.0.0-alpha.28)%3B%20autocomplete-core%20(1.0.0-alpha.28)&x-algolia-api-key=c06007c9cb270acc40e1c7799bad8326&x-algolia-application-id=BH4D9OD16A",
                "connectTimeout": 1,
                "responseTimeout": 2
            },
            "response": {
                "content": "{\"message\":\"Invalid Application-ID or API key\",\"status\":403}",
                "status": 403,
                "isTimedOut": false
            },
            "host": {
                "protocol": "https",
                "url": "BH4D9OD16A-dsn.algolia.net",
                "accept": 1
            },
            "triesLeft": 3
        }
    ]
}
```

My code:
```

          <DocSearchModal
            initialQuery={initialQuery}
            initialScrollY={window.scrollY}
            onClose={onClose}
            indexName="astro"
            apiKey="c06007c9cb270acc40e1c7799bad8326"
            appId="BH4D9OD16A"
            hitComponent={Hit}
          />,
```

Does it take a while for the index to be added?

**Comment by shortcuts** [comment]
Oopsie, here's the right api key: 0f387260ad74f9cbf4353facd29c919c

(Also, the appId is defaulted to `BH4D9OD16A` in the DocSearch component, so you don't need to pass it)

**Comment by FredKSchott** [comment]
It lives! https://astro-docs-git-wip-search-bar-pikapkg.vercel.app/getting-started
Thanks for all of your help!

**Comment by shortcuts** [comment]
Congrats!

**Pull request merged**